prow/pubsub/subscriber/subscriber.go

```diff
@@ -301,6 +350,18 @@ func (poh *postsubmitJobHandler) getProwJobSpec(cfg prowCfgClient, pc *config.In
 
 	logger := logrus.WithFields(logrus.Fields{"org": org, "repo": repo, "branch": branch, "orgRepo": orgRepo})
 	postsubmits := cfg.GetPostsubmitsStatic(orgRepo)
+	var err error
+	if poh.CookieFilePath != "" {
+		if client, ok := poh.CacheMap[orgRepo]; ok {
+			pc = client
+		} else {
+			pc, err = createCache(orgRepo, org, poh.CookieFilePath, poh.CacheSize, poh.Agent)
```

Comment:
`CacheMap` is the map of orgRepo -> `InRepoConfigCache`. CreateCache creates the `InRepoConfigCache` The `CacheMap` is initialized at the Subscriber level.

Reply:
We could probably create the `InRepoConfigCache` earlier, but not without some refactoring because we currently don't get the OrgRepo until just before the call to `CreateCache`

Reply:
for a repo to use inrepoconfig, it needs to be allow-listed in `config.yaml`, which imo probably wouldn't change in the future either. So my initial plan was to use the list of orgs and repos that are allow-listed, and initialize inrepoconfig cache when the `sub` component starts.

The challenge here is that the global config allows both `org` and `repo`, but this cache map is per repo, and we won't know that repo name until a pubsub message is received, so yeah there is not really a good way to initialize all cache ahead of time.

Another idea, how about instead of PresubmitJobHandler carries `map[string]*config.InRepoConfigCache`, it carries `type inrepoconfigFetcher struct` struct, which has `inrepoconfigCacheGetter(orgRepo string) (*config.InRepoConfigCache, error)` method, as well as CookieFilePath, CacheSize, and Agent. One advantage is that this is easier to use thread lock to avoid unsafe map manipulation in multi-thread scenario.

The other consideration, is that this cache is created for all pubsub request, even if they are not allowlisted. We should also filter these out when constructing inrepoconfig cache

Reply:
Done. Let me know if this is what you had in mind